sklearn/utils/_mocking.py

```diff
@@ -51,31 +51,45 @@ def __ne__(self, other):
 class CheckingClassifier(ClassifierMixin, BaseEstimator):
     """Dummy classifier to test pipelining and meta-estimators.
 
-    Checks some property of X and y in fit / predict.
+    Checks some property of `X` and `y`in fit / predict.
     This allows testing whether pipelines / cross-validation or metaestimators
     changed the input.
 
     Parameters
     ----------
-    check_y
-    check_X
-    foo_param
-    expected_fit_params
+    check_y, check_X : callable, default=None
+        The callable used to validate `X` and `y`.
```

Comment:
maybe indicate that these callables return bools, not the validated arrays. I was a bit confused when I reached the tests